frontends/ast/simplify.cc

```diff
@@ -2886,9 +2886,14 @@ AstNode *AstNode::readmem(bool is_readmemh, std::string mem_filename, AstNode *m
 	int meminit_size=0;
 
 	std::ifstream f;
+	std::string path = filename.substr(0, filename.find_last_of("\\/")+1);
```

Comment:
What happens here if `filename` does not contain a `/`?

Reply:
Mmm, I suppose that something will be broken XD (good catch). I will test and fix it!

Reply:
Are Windows paths considered by this new code too?

Reply:
@eddiehung I have not Windows near to check (I am a GNU/Linux user) but I did my best:
* `find_last_of("\\/")` is to take in account the both possible slashes.
* `+1` is to use the slash provided bt the filename.
* I used `std::string` (I understand that is supported in both OS)

@cliffordwolf it works also when `filename` does not contain a slash. I added `test/memfile`. With the old implementation:

```
$ ./run-test.sh 
* Creating Memory Content Files
* Running from the parent directory
  * Memory Content File: tempfile1.dat
memfile/memory.v:15: ERROR: Can not open file `tempfile1.dat` for \$readmemb.
  * Memory Content File: temp/tempfile2.dat
memfile/memory.v:15: ERROR: Can not open file `temp/tempfile2.dat` for \$readmemb.
* Running from the same directory
  * Memory Content File: tempfile1.dat
  * Memory Content File: temp/tempfile2.dat
* Running from a child directory
  * Memory Content File: tempfile1.dat
../memory.v:15: ERROR: Can not open file `tempfile1.dat` for \$readmemb.
  * Memory Content File: temp/tempfile2.dat
../memory.v:15: ERROR: Can not open file `temp/tempfile2.dat` for \$readmemb.
  * Memory Content File: tempfile2.dat
../memory.v:15: ERROR: Can not open file `temp/tempfile2.dat` for \$readmemb.
* Done
```

And with the new feature:

```
$ ./run-test.sh 
* Creating Memory Content Files
* Running from the parent directory
  * Memory Content File: tempfile1.dat
  * Memory Content File: temp/tempfile2.dat
* Running from the same directory
  * Memory Content File: tempfile1.dat
  * Memory Content File: temp/tempfile2.dat
* Running from a child directory
  * Memory Content File: tempfile1.dat
  * Memory Content File: temp/tempfile2.dat
  * Memory Content File: tempfile2.dat
* Done
```

The test is not included to be run automatically. Let me know all the needed to be accepted.

Regards

Reply:
I added it to `make test` with an output format similar to the other tests.

Reply:
> * `find_last_of("\\/")` is to take in account the both possible slashes.

Good point. I misread that.. I think it should be okay. There's no good reason why you should have a forward slash in Windows paths, or a backwards slash in Linux paths. The alternative would be an `#ifdef` to decide what the character to search for.

Reply:
I moved the path detection into the `if` where it is used:

```
	if (f.fail()) {
		std::string path = filename.substr(0, filename.find_last_of("\\/")+1);
		f.open(path + mem_filename.c_str());
		yosys_input_files.insert(path + mem_filename);
	} else {
```

Do you prefer the use of `#ifdef` as follows? 

```
	if (f.fail()) {
#ifdef _WIN32
		std::string slash = "\\";
#else
		std::string slash = "/";
#endif
		std::string path = filename.substr(0, filename.find_last_of(slash)+1);
		f.open(path + mem_filename.c_str());
		yosys_input_files.insert(path + mem_filename);
	} else {
```

Reply:
Now that you're offering it: sure! I think that can only improve things as opposed to make things worse.
The other minor nitpick is to use `char slash = ` since you really are only searching for one character as opposed to multiple.

Reply:
Done. Tested and working, at least with Linux :-D